src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -365,10 +367,34 @@ def reweight(self, svc_id, weight):
     def _create_predefined_drive_group(self, data):
         orch = OrchClient.instance()
         option = OsdDeploymentOptions(data[0]['option'])
+        nvme_model = None
+        ssd_model = None
+        for inventory_host in orch.inventory.list(hosts=None, refresh=True):
+            for device in inventory_host.devices.devices:
+                if device.available and device.human_readable_type == 'ssd':
+                    if '/dev/nvme' in device.path:
+                        nvme_model = device.sys_api['model']
+                    else:
+                        ssd_model = device.sys_api['model']
         if option in list(OsdDeploymentOptions):
             try:
                 predefined_drive_groups[
                     option]['encrypted'] = data[0]['encrypted']
+
+                if option == OsdDeploymentOptions.THROUGHPUT:
+                    if not data[0]['hdd_used']:
+                        predefined_drive_groups[
+                            option]['db_devices'].pop('rotational', 0)
+                        predefined_drive_groups[
+                            option]['data_devices'].pop('rotational', 1)
```

Comment:
Maybe worthy of a comment explaining why this? I look at this code and it's hard for me to follow...

Reply:
done!